code/controllers/subsystem/timer.dm

```diff
@@ -295,7 +295,8 @@ proc/addtimer(datum/callback/callback, wait, flags)
 		return
 
 	if (wait <= 0)
-		callback.InvokeAsync()
+		spawn(0)	//This is the one legal spawn
+			callback.Invoke()
```

Comment:
no, do what I suggested in that other pr, make it insert the timer to run this tick (wait = 0)

spawn() and sleep has a overhead of 10 or more proc calls depending on how many proc variables there are to be copied in that scope and how many other things are running in the sleep queue.

The entire point of addtimer is to be faster then spawn()